src/coprocessor/dag/expr/builtin_miscellaneous.rs

```diff
@@ -28,6 +28,48 @@ impl ScalarFunc {
         }
     }
 
+    pub fn is_ipv4_compat<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        //Not an IPv6 address, return 0
+        if input.len() != 16 {
```

Comment:
How about using a constant here?

Reply:
I have not find a constant about ipv6 length in rust, if you know please tell me...and It seems there are two ways if using a constant, one is declaring it in this file or somewhere else, another is calculate the struct in6_addr size, which one is better?

Reply:
I think declaring it in this file is ok.

Reply:
I add two constants: IPV6_LENGTH and IPV4_LENGTH, and override original number with the constant in built_miscellaneous.rs